lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/LocalStorageClient.java

```diff
@@ -184,10 +167,8 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) th
     @Override
     public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
       // Visit the file and actually process it!
-      Path fullPath = file.toAbsolutePath().normalize();
-      String fullPathStr = fullPath.toString();
-      String fileExtension = FilenameUtils.getExtension(fullPathStr);
-      processAndPublishFileIfValid(publisher, fullPathStr, fileExtension, new FileReference(fullPath, attrs.lastModifiedTime().toInstant()), params);
+      FileReference fileRef = new LocalFileReference(file, attrs.lastModifiedTime().toInstant());
```

Comment:
given that we have the attributes already here, can we pass them into the LocalFileReference instead of acquiring them a second time inside toDoc?